CrossPlatformDesktopProject/CrossPlatformDesktopProject/Link/Equipables/Bomb.cs

```diff
@@ -0,0 +1,130 @@
+﻿using CrossPlatformDesktopProject.Link;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CrossPlatformDesktopProject.Equipables
+{
+    class Bomb : IEquipable
+    {
+        private Player player;
+        private Vector2 currentPos;
+        private float usedX;
+        private float usedY;
+        private int my_frame_index;
+        private int delay_frame_index;
+        private int delay_frames;
+        private int explode_time;
+        private int current_frame;

```

Comment:
What is the difference between `current_frame` and `my_frame_index`?

Reply:
The `my_frame_index` int is used to determine which of the bomb sprites should be used from the `my_sources` list, and I used `current_frame` to determine which frame of the explosions it was on - the explosion goes back and forth between the same two frames, and both of those two frames use the same texture (`BOMB_EXPLOSION_1`), so I needed another counter to figure out which of the two "collages" could be drawn. Perhaps a better naming convention could be used there